In [1]:
# Khai báo các thư viện
import pandas as pd
import numpy as np

In [2]:
# tải tập dữ liệu
business = pd.read_csv("Raw Data.csv")

In [3]:
business.head()

country_or_area  year  comm_code                               commodity  \
0       Australia  2016      10111         Horses, live pure-bred breeding   
1       Australia  2016      10119  Horses, live except pure-bred breeding   
2       Australia  2016      10119  Horses, live except pure-bred breeding   
3       Australia  2016      10119  Horses, live except pure-bred breeding   
4       Australia  2016      10120          Asses, mules and hinnies, live   

        flow  trade_usd  weight_kg    quantity_name  quantity         category  
0     Export  128577553   900450.0  Number of items    1882.0  01_live_animals  
1  Re-Import    4928989    47240.0  Number of items     104.0  01_live_animals  
2     Export   11812782   153587.0  Number of items     276.0  01_live_animals  
3     Import   90430302  1082493.0  Number of items    2073.0  01_live_animals  
4     Export      58473     5805.0  Number of items     416.0  01_live_animals

In [4]:
# đổi tên các cột để dễ hiểu hơn
business.columns = ["Country", "Year", "Commoditycode", "Commodity", "Flow",
                    "Dollars", "Weight", "Quantityname", "Quantity", "Category"]

In [5]:
# kiểm tra cấu trúc dữ liệu
business.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 59090 entries, 0 to 59089
Data columns (total 10 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Country        59090 non-null  object 
 1   Year           59090 non-null  int64  
 2   Commoditycode  59090 non-null  int64  
 3   Commodity      59090 non-null  object 
 4   Flow           59090 non-null  object 
 5   Dollars        59090 non-null  int64  
 6   Weight         58375 non-null  float64
 7   Quantityname   59090 non-null  object 
 8   Quantity       58385 non-null  float64
 9   Category       59090 non-null  object 
dtypes: float64(2), int64(3), object(5)
memory usage: 4.5+ MB


In [6]:
# hiển thị các giá trị null
business.isnull().sum()

Country            0
Year               0
Commoditycode      0
Commodity          0
Flow               0
Dollars            0
Weight           715
Quantityname       0
Quantity         705
Category           0
dtype: int64

In [7]:
# Gán giá trị 1 cho các trường hợp Weight và Quantity có giá trị bằng 0 hoặc null
business["Missing"] = np.where((business["Weight"] == 0) | (business["Weight"].isnull())
                               | (business["Quantity"] == 0) | (business["Quantity"].isnull()), 1, 0)
print(business["Missing"].value_counts())

0    56271
1     2819
Name: Missing, dtype: int64


Có 2819 trường hợp, trong đó Weight và Quantity có giá trị bằng 0 hoặc NA

In [8]:
# Drop các trường hợp có giá trị bằng 1
business = business[business["Missing"] == 0]

In [9]:
# Xóa cột Missing
business.drop("Missing", axis=1, inplace=True)

In [10]:
#Lưu dữ liệu
business.to_csv("Cleaned Data.csv", index=False)